flytekit/clis/sdk_in_container/serialize.py

```diff
@@ -137,22 +138,12 @@ def serialize_all(project, domain, pkgs, version, folder=None):
                 continue
             serialized = entity.serialize()
             fname_index = str(i).zfill(zero_padded_length)
-            fname = "{}_{}.pb".format(fname_index, entity.id.name)
-            click.echo(f"  Writing type: {entity.id.resource_type}, {entity.id.name} to\n    {fname}")
+            fname = "{}_{}.{}.pb".format(fname_index, entity.id.name, entity.id.resource_type_name().lower())
```

Comment:
so we're embedding the name of the entity into the file name?  I wasn't going to do this because of case-sensitivity, os level constraints on file names, length restrictions and the like.  Do you really not like the separate file idea?

Reply:
from looking online, modern systems allow you up to 255 chars but i can change this to a single char `['t', 'w', 'l']` if theres a concern. the code should handle calling `.upper` when parsing. i prefer to avoid a separate identifier file when we can be concise with this alternative but if you feel strongly i'm happy to change back.

Reply:
talked offline, reverted back to using identifier files